File: zero/Maruko.Zero/Internal/Service/Menu/DTO/SysMenuDTO.cs
using System.Collections.Generic;
using Maruko.Core.Application.Servers.Dto;
using Maruko.Core.AutoMapper.AutoMapper;

namespace Maruko.Zero
{
    [AutoMap(typeof(SysMenu))]
    public class SysMenuDTO : EntityDto
    {
        public int ParentId { get; set; } = 0;
        public string Name { get; set; }
        public string Url { get; set; }
        public int Level { get; set; } = 1;
        public string Operates { get; set; }
        public string Icon { get; set; }
        public string Key { get; set; } = "";
        public bool IsLeftShow { get; set; }
        public List<OperateModel> OperateModels { get; set; } = new List<OperateModel>();
        public List<SysMenuDTO> Children { get; set; } = new List<SysMenuDTO>();
    }

    public class OperateModel : EntityDto<int>
    {
        public string Name { get; set; }
    }
}

File: act/act-modules/act-modules-zero/src/main/java/com/act/modules/zero/internal/application/menu/dto/SysMenuDTO.java
package com.act.modules.zero.internal.application.menu.dto;

import com.act.core.domain.BaseEntity;
import lombok.Data;

import java.util.ArrayList;
import java.util.List;

@Data
@SuppressWarnings("all")
public class SysMenuDTO extends BaseEntity<Long> {

    private Long parentId;

    private String name;

    private String url;

    private int level;

    private String operates;

    private int sort;

    private String icon;

    private String key;

    private int isLeftShow;

    private List<SysMenuDTO> children = new ArrayList<>();
}
